sherpa/stats/tests/test_wstat.py

```diff
@@ -0,0 +1,394 @@
+#
+#  You should have received a copy of the GNU General Public License along
+#  with this program; if not, write to the Free Software Foundation, Inc.,
+#  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+
+# Explicit tests of wstat, related to issue 248
+# https://github.com/sherpa/sherpa/issues/248
+# The intention is the functionality of these tests will be moved
+# elsewhere, but for now it is useful to have some testing of
+# the various options.
+#
+# At present the statistic tests are not broken up into unit and
+# integration tests, as most of the functionality is tested either
+# in integration tests (directly or implicitly).
+#
+# These tests are regression tests, designed to check
+# that the wstat statistic can be used with PHA data sets:
+#
+# a) one data set (backscal is a scalar)
+#   a.1) ungrouped, no filter
+#   a.2) ungrouped, energy filter
+#   a.3) ungrouped, wavelength filter
+#   a.4) grouped, no filter
+#   a.5) grouped, energy filter
+#   a.6) grouped, wavelength filter
+#
+#   The filters should be complex (e.g. multiple ignore regions)
+#   but this part is not exercised.
+#
+# b) multiple data sets (backscal is a scalar)
+#
+# Repeat a but for multiple spectra.
+#
+# c) as a but with backscal being a vector
+#
+# d) as b but with backscal being a vector
+#
+# Note that for test a a previously-grouped PHA file is used, whereas
+# for test b one of the files is grouped on the fly.
+#
+# TODO:
+#   - replace with a synthetic data set so the test can be run
+#     without data or the IO module; however, this version does test
+#     the calculation when the responses (i.e. ARF and RMF) are
+#     included.
+#
+#   - validate some/all of the statistic values to check the
+#     answer is meaningful
+#
+#   - should there be any tests that include quality flags?
+#
+#   - to test out if issue 227 is fixed
+#     https://github.com/sherpa/sherpa/issues/227
+#     try subtract/unsubtract
+#
+
+import numpy as np
+
+from sherpa.utils import SherpaTestCase, requires_data, requires_fits
+from sherpa.astro import ui
+
+from unittest import expectedFailure
+
+import logging
+
+logger = logging.getLogger('sherpa')
+
+
+@requires_data
+@requires_fits
+class test_wstat_single_scalar(SherpaTestCase):
+    """Single PHA file with a scalar backscal value"""
+
+    def setUp(self):
+
+        self._old_logger_level = logger.getEffectiveLevel()
+        logger.setLevel(logging.ERROR)
+
+        ui.set_stat('wstat')
+
+        infile = self.make_path('3c273.pi')
+        ui.load_pha(1, infile)
+
+        ui.set_source(1, ui.powlaw1d.pl)
+
+        # The powerlaw slope and normalization are
+        # intended to be "a reasonable approximation"
+        # to the data, just to make sure that any statistic
+        # calculation doesn't blow-up too much.
+        #
+        ui.set_par("pl.gamma", 1.782)
+        ui.set_par("pl.ampl", 1.622e-4)
+
+    def tearDown(self):
+        ui.clean()
```

Comment:
> I don't really understand what setting there could be which would change the statistic calculation in this case,

I guess one would need to look at the logs of the failing build and bisect the tests run before the offending test until one finds the test responsible for the failure. As I mentioned in a separate conversation, this behavior could be benign (tests should be isolated nevertheless) or uncover an underlying, maybe unrelated issue.